### PR TITLE
stretch-cluster.py: Added new test for the stretched mode 

### PR DESCRIPTION
**Pull request by pdhiran**
Test module to deploy stretched cluster on the given set of nodes from 3 Sites.

Signed-off-by: Pawan Dhiran <pdhiran@redhat.com>

### CODE REVIEW

**Comment by pdhiran** [comment]
Pass Logs : http://magna002.ceph.redhat.com/ceph-qe-logs/pawan/cephci-run-DY53GX/

The test case failed, but the feature is working as expected.

**Review by psathyan — COMMENTED** [review]
Thanks for putting this together.

We have multiple sleep statements... is it something we can avoid? Is it something we can take it later as part of technical debit?

File: tests/rados/test_deploy_stretch_cluster_baremetal.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    time.sleep(2)
+
+    # Moving the new items under root/default
+    move_crush_item(cephadm, crush_obj=site1_name, name="root", value="default")
+    time.sleep(2)
+    move_crush_item(cephadm, crush_obj=site2_name, name="root", value="default")
+    time.sleep(2)
+
+    # Moving all the OSD and Mon daemons into respective sites
+    sites = ["site1", "site2", "site3"]
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply tests/rados/test_deploy_stretch_cluster_baremetal.py:91]: These are just the python variable names that I have used in the code. these are the placeholder objects for actual site name and hosts.

File: tests/rados/test_deploy_stretch_cluster_baremetal.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    for site in sites:
+        mon_hosts = [
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply tests/rados/test_deploy_stretch_cluster_baremetal.py:93]: Done.

File: tests/rados/test_deploy_stretch_cluster_baremetal.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        return 1
+    log.info("Enabled connectivity mode on the cluster")
+
+    # Creating new datacenter crush objects
+    cmd = f"ceph osd crush add-bucket {site1_name} datacenter"
+    rados_obj.run_ceph_command(cmd)
+    time.sleep(2)
+    cmd = f"ceph osd crush add-bucket {site2_name} datacenter"
+    rados_obj.run_ceph_command(cmd)
+    time.sleep(2)
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply tests/rados/test_deploy_stretch_cluster_baremetal.py]: Done.

File: suites/pacific/rados/deploy-stretch-cluster-mode.yaml [hunk 4]
```diff
@@ -0,0 +1,116 @@
+# Suite is to be used to deploy Stretched mode on the given set of nodes
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply suites/pacific/rados/deploy-stretch-cluster-mode.yaml:1]: DOne

File: tests/rados/test_deploy_stretch_cluster_baremetal.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pdhiran) [reply tests/rados/test_deploy_stretch_cluster_baremetal.py:91]: I will have this as my backlog item and finish this with the next changes

**Comment by pdhiran** [comment]
@psathyan , I will have the suggested improvement changes in the backlog and commit them with the next changes. Let's get this in ?

**Review by psathyan — APPROVED** [review]

**Pull request merged**